CardsCashCasino/BlackjackManager.cs

```diff
@@ -0,0 +1,215 @@
+        /// Starts a new game by generating and shuffling the deck, then dealing initial cards.
+        /// </summary>
+        public void StartNewGame()
+        {
+            _playerHand.Clear();
+            _playerSplitHand.Clear();
+            _dealerHand.Clear();
+            _gameOver = false;
+            _splitHand = false;
+            _cardManager.ClearDecks();
+            _cardManager.GenerateDecks(1);
+            _playerHand.Add(_cardManager.DrawCard());
+            _dealerHand.Add(_cardManager.DrawCard());
+            _playerHand.Add(_cardManager.DrawCard());
+            _dealerHand.Add(_cardManager.DrawCard());
+        }
+
+        public int GetBlackjackValue(List<Card> hand)
+        {
+            int value = 0;
+            int aceCount = 0;
+
+            foreach (Card card in hand)
+            {
+                switch (card.Value)
+                {
+                    case Value.Two:
+                    case Value.Three:
+                    case Value.Four:
+                    case Value.Five:
+                    case Value.Six:
+                    case Value.Seven:
+                    case Value.Eight:
+                    case Value.Nine:
+                    case Value.Ten:
+                        value += (int)card.Value; // Number cards are worth their face value
+                        break;
+                    case Value.Jack:
+                    case Value.Queen:
+                    case Value.King:
+                        value += 10; // Face cards are worth 10 points each
+                        break;
+                    case Value.Ace:
+                        aceCount++; // Track the number of aces
+                        break;
+                }
+            }
+
+            // Add ace values, either as 11 or 1, as appropriate
+            for (int i = 0; i < aceCount; i++)
+            {
+                if (value + 11 <= 21)
+                {
+                    value += 11; // Ace is worth 11 if it doesn’t push over 21
+                }
+                else
+                {
+                    value += 1; // Otherwise, Ace is worth 1
+                }
+            }
+
+            return value;
+        }
+        public void SplitHand()
+        {
+            // Check if the player has exactly two cards of the same value
+            if (_playerHand.Count == 2 && _playerHand[0].Value == _playerHand[1].Value)
+            {
+                // Move the second card to the split hand
+                _playerSplitHand.Add(_playerHand[1]);
+                _playerHand.RemoveAt(1);
+
+                // Draw a new card for each hand
+                _playerHand.Add(_cardManager.DrawCard());
+                _playerSplitHand.Add(_cardManager.DrawCard());
+            }
+        }
+        /// <summary>
+        /// Determines if the player has won, lost, or if the game is still ongoing.
+        /// </summary>
+        public string GetGameStatus(List<Card> hand)
```

Comment:
You'll have to make sure to set the _gameOver flag to True for each else if block